### PR TITLE
LibWeb+Ladybird+Userland: Port window.[alert,confirm,prompt] to String

### PR DESCRIPTION
**Pull request by trflynn89**

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/Window.cpp [hunk 1]
```diff
@@ -1105,20 +1105,16 @@ bool Window::confirm(Optional<String> const& message)
...
 {
     // FIXME: Make this fully spec compliant.
-    if (auto* page = this->page()) {
-        auto response = page->did_request_prompt(message->to_deprecated_string(), default_->to_deprecated_string());
-        if (response.is_null())
-            return {};
-        return String::from_deprecated_string(response).release_value_but_fixme_should_propagate_errors();
-    }
+    if (auto* page = this->page())
+        return page->did_request_prompt(*message, *default_);
```
Comment (konradekk) [inline Userland/Libraries/LibWeb/HTML/Window.cpp:1117]: 🤯

**Review by konradekk — COMMENTED** [review]
looks neat!

File: Userland/Libraries/LibWeb/HTML/Window.cpp (hunk 1 above)
Comment (GMTA) [reply Userland/Libraries/LibWeb/HTML/Window.cpp:1117]: @konradekk Please refrain from posting nonsensical comments like these, we need a review instead!

**Review by linusg — APPROVED** [review]
Looks great :)

File: Userland/Libraries/LibWeb/HTML/Window.cpp (hunk 1 above)
Comment (konradekk) [reply Userland/Libraries/LibWeb/HTML/Window.cpp:1117]: i just really liked what it became (seeing the previous version side-by-side).

but sure—got that. 😣

**Pull request merged**